plugin.video.101barz/default.py

```diff
@@ -0,0 +1,247 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+from __future__ import division
+from future import standard_library
+standard_library.install_aliases()
+from builtins import str
+import urllib.request, urllib.parse, urllib.error
+import xbmcplugin
+import xbmcgui
+import sys
+import xbmcaddon
+import xbmc
+import os
+import requests
+import json
+
+ADDON = "plugin.video.101barz"
+SETTINGS = xbmcaddon.Addon(id=ADDON)
+LANGUAGE = SETTINGS.getLocalizedString
+IMAGES_PATH = os.path.join(xbmcaddon.Addon(id=ADDON).getAddonInfo('path'), 'resources', 'images')
+PLUGIN_HANDLE = int(sys.argv[1])
+BASE_URL = "https://101barz.bnnvara.nl/api/video?"
+HEADERS = {'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10115) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/50.0.2661.102 Safari/537.36'}
+COOKIES = {}
+DATE = "2019-02-12"
+VERSION = "1.0.0"
+
+SEARCHURL = "category=<categorystring>&search=<searchstring>&offset=0&limit=12"
+ALLESURL = "category=all&search=&offset=0&limit=12"
+STUDIOSESSIONSURL = "category=13&search=&offset=0&limit=12"
+VIDEOCLIPZURL = "category=14&search=&offset=0&limit=12"
+ROTJOCHLOUNGEURL = "category=25&search=&offset=0&limit=12"
+HITMAKERZURL = "category=27&search=&offset=0&limit=12"
+BARZ4BARZURL = "category=28&search=&offset=0&limit=12"
+TALENTVDMAANDURL = "category=29&search=&offset=0&limit=12"
+
+
+def index():
+    add_dir(LANGUAGE(30002), ALLESURL, 'list_videos', "")
+    add_dir(LANGUAGE(30003), STUDIOSESSIONSURL, 'list_videos', "")
+    add_dir(LANGUAGE(30004), VIDEOCLIPZURL, 'list_videos', "")
+    add_dir(LANGUAGE(30005), ROTJOCHLOUNGEURL, 'list_videos', "")
+    add_dir(LANGUAGE(30006), HITMAKERZURL, 'list_videos', "")
+    add_dir(LANGUAGE(30007), BARZ4BARZURL, 'list_videos', "")
+    add_dir(LANGUAGE(30008), TALENTVDMAANDURL, 'list_videos', "")
+
+    xbmcplugin.endOfDirectory(PLUGIN_HANDLE)
+
+
+def list_videos(url):
+    log("url", url)
+
+    # add category number to search url
+    CATEGORY = "category="
+    start_pos = convertToUnicodeString(url).find(CATEGORY)
+    if start_pos >= 0:
+        end_pos = convertToUnicodeString(url).find("&", start_pos)
+        category_value = convertToUnicodeString(url)[start_pos + len(CATEGORY):end_pos]
+
+        log("category_value", category_value)
+
+        search_url = convertToUnicodeString(SEARCHURL).replace("<categorystring>", category_value)
+
+        log("search_url", search_url)
+
+        # add search url
+        add_dir(LANGUAGE(30001), search_url, 'search', "")
+
+    html_source = get_url(url)
+    json_data = json.loads(html_source)
+
+    log("json_data", json_data)
+
+    for item in json_data['video']:
+        log("item", item)
+
+        length = ""
+        title = item['title']
+        desc = item['subtitle']
+        date = item['publish_up']
+        thumb = item['card_image']
+        youtube_id = item['youtube_id']
+        youtube_url = 'plugin://plugin.video.youtube/play/?video_id=%s' % youtube_id
+
+        log("youtube_url", youtube_url)
```

Comment:
already logged above in the json_data and item log line

Reply:
I will remove this log.